### PR TITLE
restructure of geometry

### PR DESCRIPTION
**Pull request by wkerzendorf**

### CODE REVIEW

**Comment by wkerzendorf** [comment]
It's draft for now but good to follow

File: tardis/model/geometry/tests/test_radial1d.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    assert homologous_radial1d_geometry.v_inner_boundary_index == 0
+    assert homologous_radial1d_geometry.v_outer_boundary_index == 12
+    vib_index = homologous_radial1d_geometry.v_inner_boundary_index
+    vob_index = homologous_radial1d_geometry.v_outer_boundary_index
+    assert np.all(
+        homologous_radial1d_geometry.v_inner[vib_index:vob_index]
+        == homologous_radial1d_geometry.v_inner
+    )
+
+    # pivoting around the inner boundary of the simulation
```
Comment (wkerzendorf) [inline tardis/model/geometry/tests/test_radial1d.py:40]: @andrewfullard @Rodot- - can you see if that logic is correct

File: tardis/model/geometry/radial1d.py [hunk 2]
```diff
@@ -21,17 +21,79 @@ class Radial1DGeometry:
...
+        self.v_outer_boundary = v_outer_boundary
+
+    @property
+    def v_inner_boundary_index(self):
+        # TODO potentially rename to v_inner_active_index??
+
+        # fix to ensure that we get the same index if we are on the shell
+        # boundary vs slightly above
+
+        return np.clip(
```
Comment (andrewfullard) [inline tardis/model/geometry/radial1d.py:60]: Why is np.clip being used here?

File: tardis/model/geometry/tests/test_radial1d.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    # boundary are on the innermost and outermost shell
+
+    homologous_radial1d_geometry.v_inner_boundary = (
+        homologous_radial1d_geometry.v_inner[0]
+    )
+    homologous_radial1d_geometry.v_outer_boundary = (
+        homologous_radial1d_geometry.v_outer[-1]
+    )
+    assert homologous_radial1d_geometry.v_inner_boundary_index == 0
+    assert homologous_radial1d_geometry.v_outer_boundary_index == 12
```
Comment (andrewfullard) [inline tardis/model/geometry/tests/test_radial1d.py:32]: I think these values should be parametrized in some way, or made more clear where they come from

File: tardis/model/geometry/tests/test_radial1d.py (hunk 1 above)
Comment (andrewfullard) [reply tardis/model/geometry/tests/test_radial1d.py:40]: Makes sense to me, though the logic here shows that your implementation does not report any error if something is "inside" the inner boundary or "outside" the outer boundary. Is that intended?

File: tardis/model/geometry/radial1d.py (hunk 2 above)
Comment (wkerzendorf) [reply tardis/model/geometry/radial1d.py:60]: so we don't go negative in the index.

File: tardis/model/geometry/tests/test_radial1d.py (hunk 1 above)
Comment (wkerzendorf) [reply tardis/model/geometry/tests/test_radial1d.py:40]: that is an easy fix - for now. I would just think that we keep things the same. So if we have 8000 - 9000 - density = 1e-13 - then v_inner_boundary at 7500, then the new cell will be 7500 - 9000 with density 1e-13.

Comment (andrewfullard) [reply tardis/model/geometry/tests/test_radial1d.py:43]: Okay. I assume this does *not* affect packets going within the inner boundary or leaving the outer boundary then?

File: tardis/model/base.py [hunk 4]
```diff
@@ -214,18 +214,20 @@ def __init__(
         v_boundary_outer=None,
         electron_densities=None,
     ):
+        self.geometry = geometry
         self._v_boundary_inner = None
         self._v_boundary_outer = None
-        self._velocity = None
-        self.raw_velocity = velocity
+        #self._velocity = None
```
Comment (andrewfullard) [inline tardis/model/base.py:220]: Delete commented lines?

File: tardis/model/geometry/radial1d.py [hunk 5]
```diff
@@ -21,17 +21,105 @@ class Radial1DGeometry:
         Volume in each shell
     """
 
-    def __init__(self, r_inner, r_outer, v_inner, v_outer):
-        self.r_inner = r_inner
-        self.r_outer = r_outer
-        self.v_inner = v_inner
-        self.v_outer = v_outer
+    DEFAULT_VELOCITY_UNIT = u.km / u.s
+    DEFAULT_DISTANCE_UNIT = u.km
```
Comment (andrewfullard) [inline tardis/model/geometry/radial1d.py:25]: Not `u.cm`?

**Comment by atharva-2001** [comment]
Might be helpful- rebasing this or merging the master branch with this would disable code coverage workflow from checking on tests, so it would fail only when necessary.

File: tardis/model/geometry/tests/test_radial1d.py (hunk 1 above)
Comment (wkerzendorf) [reply tardis/model/geometry/tests/test_radial1d.py:43]: nope

File: tardis/model/base.py [hunk 6]
```diff
@@ -308,12 +299,17 @@ def __init__(
...
+                    / constants.c
+                )
             )
-        elif len(t_radiative) != self.no_of_shells:
+        
+        elif len(t_radiative) == self.no_of_shells + 1:
             t_radiative = t_radiative[
-                self.v_boundary_inner_index
-                + 1 : self.v_boundary_outer_index
+                self.geometry.v_inner_boundary_index
```
Comment (Rodot-) [inline tardis/model/base.py:311]: Change v_inner_boundary_index to inner_boundary_index

File: tardis/model/geometry/radial1d.py [hunk 7]
```diff
@@ -21,17 +22,140 @@ class Radial1DGeometry:
...
+    def v_inner_boundary_index(self):
+        return np.clip(
+            np.searchsorted(self.v_inner, self.v_inner_boundary, side="right")
+            - 1,
+            0,
+            None,
+        )
+
+    @property
+    def v_outer_boundary_index(self):
```
Comment (Rodot-) [inline tardis/model/geometry/radial1d.py:89]: Change name to outer_boundary_index

File: tardis/model/base.py (hunk 6 above)
Comment (wkerzendorf) [reply tardis/model/base.py:303]: can we make that an issue for now?

File: tardis/model/geometry/radial1d.py (hunk 7 above)
Comment (wkerzendorf) [reply tardis/model/geometry/radial1d.py:89]: same as above.

File: tardis/model/tests/test_base.py [hunk 8]
```diff
@@ -78,9 +79,9 @@ def setup(self, example_model_file_dir):
         self.simulation_state = SimulationState.from_config(self.config)
 
     def test_velocities(self):
-        assert self.simulation_state.v_inner.unit == u.Unit("cm/s")
```
Comment (andrewfullard) [inline tardis/model/tests/test_base.py:81]: I notice that the default unit appeared to be cm/s. See previous comment regarding km. Not sure if we want to "pick" one or the other as long as we convert properly.

File: tardis/montecarlo/montecarlo_numba/tests/test_base.py [hunk 9]
```diff
@@ -33,6 +33,7 @@ def test_montecarlo_main_loop(
     config_montecarlo_1e5_verysimple.montecarlo.no_of_virtual_packets = 0
     config_montecarlo_1e5_verysimple.montecarlo.iterations = 1
     config_montecarlo_1e5_verysimple.plasma.line_interaction_type = "macroatom"
+
```
Comment (andrewfullard) [inline tardis/montecarlo/montecarlo_numba/tests/test_base.py:36]: ?

File: tardis/montecarlo/montecarlo_numba/tests/test_vpacket.py [hunk 10]
```diff
@@ -66,7 +66,8 @@ def test_trace_vpacket_within_shell(
     )
 
     npt.assert_almost_equal(tau_trace_combined, 8164850.891288479)
-    npt.assert_almost_equal(distance_boundary, 843684056256104.1)
+    # changed from almost equal to allclose. Now seems to work.
```
Comment (andrewfullard) [inline tardis/montecarlo/montecarlo_numba/tests/test_vpacket.py:69]: I assume you don't know why

File: tardis/simulation/tests/test_simulation.py [hunk 11]
```diff
@@ -112,6 +112,9 @@ def test_plasma_estimates(simulation_one_loop, refdata, name):
 def test_plasma_state_iterations(simulation_one_loop, refdata, name):
     actual = getattr(simulation_one_loop, name)
 
+    if hasattr(actual, "value"):
+        actual = actual.value
```
Comment (andrewfullard) [inline tardis/simulation/tests/test_simulation.py:116]: This strips units off?

Comment (andrewfullard) [reply tardis/simulation/tests/test_simulation.py:116]: Could use a comment about why it is here

File: tardis/model/base.py [hunk 12]
```diff
@@ -391,22 +377,33 @@ def t_radiative(self, value):
 
     @property
     def radius(self):
+        return None
```
[the remark this answers is not in the document]
Comment (wkerzendorf) [reply tardis/model/base.py:380]: Was a test to remove. Should we remove? Or should it mirror `.velocity`

File: tardis/model/base.py [hunk 13]
```diff
@@ -391,22 +377,33 @@ def t_radiative(self, value):
...
+        return self.model_state.geometry.r_outer_active
 
     @property
     def r_middle(self):
         return 0.5 * self.r_inner + 0.5 * self.r_outer
 
     @property
     def velocity(self):
+        velocity = self.geometry.v_outer_active.copy()
+        return velocity.insert(0, self.geometry.v_inner_active[0])
```
[the remark this answers is not in the document]
Comment (wkerzendorf) [reply tardis/model/base.py:406]: will clean up.

File: tardis/model/tests/test_base.py (hunk 8 above)
Comment (wkerzendorf) [reply tardis/model/tests/test_base.py:81]: remove defaults?

File: tardis/simulation/tests/test_simulation.py (hunk 11 above)
Comment (wkerzendorf) [reply tardis/simulation/tests/test_simulation.py:116]: yes - it gets converted to pandas Series - and that doesn't work with the new pandas

File: tardis/model/geometry/radial1d.py (hunk 5 above)
Comment (andrewfullard) [reply tardis/model/geometry/radial1d.py:26]: I think setting the defaults to be cgs makes most sense

File: tardis/model/tests/test_base.py (hunk 8 above)
Comment (andrewfullard) [reply tardis/model/tests/test_base.py:81]: cgs defaults make sense to me

File: tardis/model/geometry/tests/test_radial1d.py (hunk 3 above)
Comment (wkerzendorf) [reply tardis/model/geometry/tests/test_radial1d.py]: this one says outdated for me.

File: tardis/montecarlo/montecarlo_numba/tests/test_vpacket.py (hunk 10 above)
Comment (wkerzendorf) [reply tardis/montecarlo/montecarlo_numba/tests/test_vpacket.py:69]: no - you think it's fine?

File: tardis/simulation/tests/test_simulation.py (hunk 11 above)
Comment (wkerzendorf) [reply tardis/simulation/tests/test_simulation.py:117]: This is added but it doesn't show up - no idea why - see latest commit

File: tardis/model/tests/test_base.py (hunk 8 above)
Comment (wkerzendorf) [reply tardis/model/tests/test_base.py:81]: but we are converting - so there is nothing to be done here - correct?

File: tardis/montecarlo/montecarlo_numba/tests/test_vpacket.py (hunk 10 above)
Comment (andrewfullard) [reply tardis/montecarlo/montecarlo_numba/tests/test_vpacket.py:69]: Presumably it is related to the data types. The comment is fine anyway

**Review by andrewfullard — DISMISSED** [review]

File: tardis/montecarlo/montecarlo_numba/tests/test_base.py (hunk 9 above)
Comment (wkerzendorf) [reply tardis/montecarlo/montecarlo_numba/tests/test_base.py:36]: not sure what you mean by this.

**Review by andrewfullard — APPROVED** [review]

**Review by Rodot- — APPROVED** [review]

**Pull request merged**